lib/util/util.js

```diff
@@ -764,6 +764,26 @@ const Util = {
         const options = { stdio: [0, 1, 2] };
         return child_process.execSync(cmd + ' ' + args.join(' '), options);
     },
+    /**
+     * helper to standardize retrieveTemplate result checks
+     * @param {Array} results array of metadata
+     * @param {string} keyToSearch the field which contains the searched value
+     * @param {string} searchValue the value which is being looked for
+     * @returns {Object} metadata to be used in building template
```

Comment:
1st Param and returns could be typed better. We have definition for MetadataTypeItem

Reply:
Still a bit new to me to use the types like this :) let me know if it works as this

Reply:
i would usually use MetadataTypeItem[] here instead tbh but I'm not sure it works because of our missing imports. maybe we need to move the declaration into util.js
let me play with it...

Reply:
Cannot be covered right now, to be reassessed later with typescript implementation